main.go

```diff
@@ -34,13 +53,26 @@ func main() {
 
 	// We define a server factory to create the custom server once all command
 	// line flags are parsed and all microservice configuration is storted out.
-	newServerFactory := func() microserver.Server {
+	newServerFactory := func(v *viper.Viper) microserver.Server {
 		// Create a new custom service which implements business logic.
 		var newService *service.Service
 		{
 			serviceConfig := service.DefaultConfig()
 
+			serviceConfig.Flag = f
 			serviceConfig.Logger = newLogger
+			serviceConfig.Viper = v
+
+			k8sTlsClientConfig := k8sclient.TLSClientConfig{
+				CertFile: Flags.Kubernetes.TLS.CrtFile,
+				KeyFile:  Flags.Kubernetes.TLS.KeyFile,
+				CAFile:   Flags.Kubernetes.TLS.CAFile,
+			}
+			serviceConfig.K8sConfig = k8sclient.Config{
+				Address:         Flags.Kubernetes.Address,
```

Comment:
The flags are not used like this with the viper configuration. Check Kubernetesd. It should look like this. 

```
config.Viper.GetString(config.Flag.Service.Kubernetes.Address)
```

Lets talk if this is kind of unclear.